Realm/Tests/MigrationTests.mm

```diff
@@ -243,6 +243,34 @@ - (void)testGetSchemaVersion {
     XCTAssertEqual(1U, [RLMRealm schemaVersionAtURL:config.fileURL encryptionKey:nil error:nil]);
 }
 
+- (void)testGetSchemaVersionOnReadOnlyRealm {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSURL *directoryURL = [RLMDefaultRealmURL() URLByDeletingLastPathComponent];
+    
+    NSError *error = nil;
+    NSURL *readOnlyDirectoryURL = [directoryURL URLByAppendingPathComponent:@"ReadOnly"];
+    XCTAssertTrue([fileManager createDirectoryAtURL:readOnlyDirectoryURL withIntermediateDirectories:YES attributes:nil error:&error]);
+    XCTAssertNil(error);
+    
+    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
+    config.fileURL = [readOnlyDirectoryURL URLByAppendingPathComponent:@"default.realm"];
+    @autoreleasepool { [RLMRealm realmWithURL:config.fileURL]; }
+    XCTAssertEqual(0U, [RLMRealm schemaVersionAtURL:config.fileURL encryptionKey:nil error:nil]);
+    
+    error = nil;
+    XCTAssertTrue([fileManager setAttributes:@{NSFilePosixPermissions: @0x100} ofItemAtPath:config.fileURL.path error:&error]);
+    XCTAssertTrue([fileManager setAttributes:@{NSFilePosixPermissions: @0x100} ofItemAtPath:readOnlyDirectoryURL.path error:&error]);
```

Comment:
Ah, I missed those lines when skimming the test. The attempt to remove the files appears to be failing due to the restrictive permissions the tests set on the directories. See https://travis-ci.org/realm/realm-cocoa/jobs/127903409, for instance, which says:

> /Users/travis/build/realm/realm-cocoa/Realm/Tests/MigrationTests.mm:271: error: -[MigrationTests testGetSchemaVersionOnReadOnlyRealm] : ((error) == nil) failed: "Error Domain=NSCocoaErrorDomain Code=513 "“ReadOnly” couldn’t be removed because you don’t have permission to access it." UserInfo={NSFilePath=/Users/travis/Library/Application Support/io.realm.TestHost/ReadOnly, NSUserStringVariant=(